include/universal/blas/generators/randsvd.hpp

```diff
@@ -39,27 +36,80 @@ namespace sw {
                 }
                 return rank;
             }
+            template<typename Matrix, typename MatrixQ, typename MatrixR, typename Scalar>
+            void qr(const Matrix& A, MatrixQ& Q, MatrixR& R) {
+                typedef typename matrix<Scalar>::value_type     value_type;//use Scalar

```

Comment:
don't use typedef's in modern C++, instead use using:

using value_type = typename matrix<Scalar>::value_type;